.github/workflows/deploy-main.yml

```
@@ -0,0 +1,28 @@
+name: deploy-main
+
+on:
+  push:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Python dependencies
+        run: python3 -m pip install -r ./requirements.txt
+
+      - name: Configure Git user
+        run: |
+          git config --local user.email "${GITHUB_ACTOR}@users.noreply.${GITHUB_DOMAIN:-"github.com"}"
+          git config --local user.name "${GITHUB_ACTOR}"
+
+      - name: Configure Git user
+        run: git fetch origin gh-pages --depth=1
+
+      - name: Deploy with mike 🚀
+        run: |
+          mike deploy compliantkubernetes ck8s -t 'main'
```

Can we write something like below?

```suggestion
          if [ "$GITHUB_REF_NAME" == "main" ]; then
              # Latest version of the docs goes under https://elastisys.io/compliantkubernetes
              VERSION_TITLE="main"
              VERSION="compliantkubernetes"
          else
              # Other versions go under https://elastisys.io/compliantkubernetes-v0.25/
              VERSION_TITLE=$(echo "$GITHUB_REF_NAME" | cut -d- -f2)
              VERSION="compliantkubernetes-${VERSION_TITLE}"
          fi
          mike deploy $VERSION_TITLE ck8s -t $VERSION
          mike set-default compliantkubernetes --push
```

I sure I missed something. :smile:

Seems to me like the branch name is easier to access via [GitHub Action Environment Variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables).

---

I only found this before `GITHUB_REF`, but i see now that `GITHUB_HEAD_REF` would work :slightly_smiling_face: 
Do you want the tag to be "compliantkubernetes-${VERSION_TITLE}"? I had it to just be ${VERSION_TITLE}. 

I split it up as to was much simpler :wink:

---

But i can merge them into one like you suggested here

---

> Do you want the tag to be "compliantkubernetes-${VERSION_TITLE}"?

Yes, please.

We decided for the URL of other versions to be `https://elastisys.io/compliantkubernetes-v0.25`, so making the tag `compliantkubernetes-${VERSION_TITLE}` is an easy way to achieve that.

---

@cristiklein I have now created the new github action for my fork. You can look at it here https://robinawallace.github.io/compliantkubernetes/compliantkubernetes

---

Cool! I check and the version name and slug is perfect. However, the URL contains `compliantkubernetes` twice. Can we remove the extra `compliantkubernetes` or is that just an artifact of your test?

---

That is just an artifact of using githubs url, the first compliantkubernetes is just the repos name. But great i will copy over my changes then :+1: